### PR TITLE
Login for discuss

### LINKED ISSUE #2224: Login / Logout for discuss.webknossos.org is buggy
While login to discuss.webknossos.org works in general, there are some usability issues:

- [ ] One can not log out from discuss itself. You have to logout from wK and then be auto logged off from discuss as well.
- [x] When logged out, navigating to discuss.webknossos.org shows you the `401` error message. There is no way / redirect to login.

Perhaps, we can find a way around these issues within the new Silhouette framework.

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2224/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by rschwanhold**
### Mailable description of changes:
 - Previously you got a 401 error when trying to access discuss.webknossos.org while not being logged in on webknossos itself.
- Now you get redirect to login and after that you get redirected back to discuss.

### Steps to test:
- try to access discuss.webknossos.org while being logged in on webknossos
- try to access discuss.webknossos.org while not being logged in on webknossos

### Issues:
- fixes #2224

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/admin/views/auth/login_view.js [hunk 1]
```diff
@@ -24,9 +25,15 @@ class LoginView extends React.PureComponent<Props> {
     this.props.form.validateFields(async (err: ?Object, formValues: Object) => {
       if (!err) {
         await Request.sendJSONReceiveJSON("/api/auth/login", { data: formValues });
-        const user = await getActiveUser();
-        Store.dispatch(setActiveUserAction(user));
-        this.props.history.push("/dashboard");
+        if (!Utils.hasUrlParam("redirectPage")) {
```
Comment (hotzenklotz) [inline app/assets/javascripts/admin/views/auth/login_view.js]: Nice javascript :-)

File: app/assets/javascripts/admin/views/auth/login_view.js [hunk 2]
```diff
@@ -24,9 +25,15 @@ class LoginView extends React.PureComponent<Props> {
...
+          const user = await getActiveUser();
+          Store.dispatch(setActiveUserAction(user));
+          this.props.history.push("/dashboard");
+        } else {
+          const user = await getActiveUser();
+          Store.dispatch(setActiveUserAction(user));
+          window.location.replace(Utils.getUrlParamValue("redirectPage"));
+        }
```
Comment (hotzenklotz) [inline app/assets/javascripts/admin/views/auth/login_view.js]: In the case were we only want to redirect there is no need to request a user. Hence, delete lines 33 & 34.

**Comment by hotzenklotz** [comment]
@rschwanhold Have you been able to test this? How can I test this?

@fm3 Can you please review the scala changes.

Much security. So wow.
giphy 1

**Comment by rschwanhold** [comment]
@hotzenklotz I tested it with by using the sso-route. "http://localhost:9000/api/auth/sso?sso=test&sig=test2" should either redirect you to login or say "invalid signature" because "test" and "test2" are just random values.

**Pull request merged**